Upgraded PagerDuty integration to version 2.

ronykoz opened the pull request:
Fix https://github.com/demisto/etc/issues/10471

ronykoz commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -0,0 +1,475 @@
+  isfetch: true
+  runonce: false
+hidden: false
+fromversion: 3.5.0
```
Not sure which version should i fill in

ronykoz replied:
OK, removed

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -0,0 +1,475 @@
+  defaultvalue: ""
+  type: 0
+  required: false
+- display: Fetch incidents
```
@ronykoz we are not using fetch here, you can take this off

ronykoz replied:
Done

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -0,0 +1,475 @@
+    - contextPath: PagerDuty.Event
+      description: Event data
+    description: Creates a new event/incident in PagerDuty
+  isfetch: true
```
No fetch, unclick the `fetches incidents` button

ronykoz replied:
Done

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -0,0 +1,475 @@
+                          'Authorization': 'Token token=' + API_KEY,
+                          'Accept': 'application/vnd.pagerduty+json;version=2'
+                      }
+
```
@ronykoz very organized, good job!

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -0,0 +1,475 @@
+    import datetime
+
+    ''' GLOBAL VARS '''
+    USE_SSL = True
```
@ronykoz do we **always** use SSL? Usually, it is a configurable parameter

ronykoz replied:
Yes, this is from the their documentation.
```To prevent accidentally sending sensitive information over the Internet, API v2 requests require SSL. Connections made over HTTP will be refused.```

hirshbergshachar replied:
@ronykoz ok, please add a comment in the code that clarifies that they enforce SSL usage. (to prevent future unnecessary changes)

hirshbergshachar replied:
@ronykoz please change to:
`# Pagerduty API works only with secured communication.`

ronykoz replied:
Done

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -0,0 +1,475 @@
+
+    def extract_on_call_user_data(users):
+        """Extact data about user from a given schedule."""
+        outputs = []
```
@ronykoz it seems that outputs and context are identical here. If so we can leave just one of them and use it both for `Contents` and `EntryContext`

ronykoz replied:
Done

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -0,0 +1,475 @@
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(USERS_ON_CALL, outputs, USERS_ON_CALL_NOW_HEADERS),
+            'EntryContext': {
+                'PagerDutyUser': context
```
@ronykoz did you make sure that the context is the same as in the old integration? Please run the commands one time in the old and one time in the new and make sure it's the same, let's make sure we are backwards compatible.

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -0,0 +1,475 @@
+
+    def extract_on_call_now_user_data(oncalls, table_name):
+        """Extract the user data from the oncalls json."""
+        outputs = []
```
@ronykoz it seems that outputs and context are identical here. If so we can leave just one of them and use it both for `Contents` and `EntryContext`

ronykoz replied:
Done

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -0,0 +1,475 @@
+
+    def extract_incidents_data(incidents):
+        """Extact data about incidents."""
+        outputs = []
```
@ronykoz it seems that outputs and context are identical here. If so we can leave just one of them and use it both for `Contents` and `EntryContext`

ronykoz replied:
Done

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -0,0 +1,475 @@
+    def extract_all_schedules_data(schedules):
+        """Extract the data about all the schedules."""
+        outputs = []
+        context = []
```
@ronykoz it seems that outputs and context are identical here. If so we can leave just one of them and use it both for `Contents` and `EntryContext`

ronykoz replied:
Done

ronykoz commented:
@hirshbergshachar Uploaded another version of the code, fixed the comments.
Several changes i've made:

- The original integration includes optional password validation, can't find where they use it, is it needed?
- Removed subdomain input, not needed in the new api
- get_users_on_call_now_command does not support query anymore

And we should open an issue fixing the misspell at the PagerDutyGetUsersOnCall outputs

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -124,16 +122,18 @@ script:
             'HumanReadable': tableToMarkdown(USERS_ON_CALL, outputs, USERS_ON_CALL_NOW_HEADERS),
             'EntryContext': {
-                'PagerDutyUser': context
+                'PagerDutyUser(true)': contexts
```
@ronykoz why is the condition equals to `true`? should be `PagerDutyUser(val.ID==obj.ID)`

ronykoz replied:
Wanted it to overwrite all the previous data

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -157,14 +163,13 @@ script:
             'HumanReadable': tableToMarkdown(table_name, outputs, USERS_ON_CALL_NOW_HEADERS),
             'EntryContext': {
-                'PagerDutyUser': context
+                'PagerDutyUser(true)': contexts
```
@ronykoz why is the condition equals to true? should be `PagerDutyUser(val.ID==obj.ID)`

ronykoz replied:
Same as above

hirshbergshachar commented on Integrations/integration-PagerDuty.yml:
```diff
@@ -246,11 +260,11 @@ script:
             'HumanReadable': tableToMarkdown(SCHEDULES, outputs, SCHEDULES_HEADERS),
             'EntryContext': {
-                'PagerDuty.Schedules': context
+                'PagerDuty.Schedules': contexts
```
@ronykoz should be `PagerDuty.Schedules(val.id==obj.id)`

ronykoz replied:
Same as above

ronykoz commented on TestPlaybooks/playbook-PagerDuty_Test.yml:
```diff
@@ -0,0 +1,226 @@
+id: 08ebe288-27a0-4631-8130-35592c22620f
+version: 5
```
Done

ronykoz commented on TestPlaybooks/playbook-PagerDuty_Test.yml:
```diff
@@ -0,0 +1,226 @@
+id: 08ebe288-27a0-4631-8130-35592c22620f
```
Done
